src/object/cli_mod.c

```diff
@@ -12,21 +12,110 @@
 #include <daos/common.h>
 #include <daos/rpc.h>
 #include <daos/mgmt.h>
+#include <daos/tls.h>
+#include <daos/job.h>
+#include <gurt/telemetry_common.h>
+#include <gurt/telemetry_producer.h>
 #include <daos_types.h>
 #include "obj_rpc.h"
 #include "obj_internal.h"
 
 unsigned int	srv_io_mode = DIM_DTX_FULL_ENABLED;
 int		dc_obj_proto_version;
 
+static void*
+dc_obj_tls_init(int tags, int xs_id, int pid)
+{
+	struct dc_obj_tls *tls;
+	int		  opc;
+	int		  rc;
+	unsigned long	  tid = pthread_self();
+
+	D_ALLOC_PTR(tls);
+	if (tls == NULL)
+		return NULL;
+
+	/** register different per-opcode sensors */
+	for (opc = 0; opc < OBJ_PROTO_CLI_COUNT; opc++) {
+		/** Start with number of active requests, of type gauge */
+		rc = d_tm_add_metric(&tls->cot_op_active[opc], D_TM_STATS_GAUGE,
+				     "number of active object RPCs", "ops",
+				     "%s/%u/%lu/ops/%s/active", dc_jobid, pid, tid,
```

Comment:
Please put these ops metrics under `io/`, for consistency with the server-side metric paths. This makes it easier to reuse code for the Prometheus exporter.
```suggestion
				     "%s/%u/%lu/io/ops/%s/active", dc_jobid, pid, tid,
```

Reply:
sure